crates/sol-macro-expander/src/expand/mod.rs

```diff
@@ -884,18 +918,32 @@ fn expand_event_tokenize<'a>(
             .filter(|(_, p)| !p.is_indexed())
             .map(|(i, p)| (i, &p.ty, p.name.as_ref())),
         cx,
+        params_len,
+        retain_fields,
     )
 }
 
+/// The `retain_fields` parameter specifies whether fields should retained as is and the struct
+/// should not be converted to unit/tuple struct i.e maintain the original form of the struct.
+/// Used to handle expansion of solidity structs as we want to retain the fields and keep original
+/// form.
 fn tokenize_<'a>(
     iter: impl Iterator<Item = (usize, &'a Type, Option<&'a SolIdent>)>,
     cx: &'a ExpCtxt<'_>,
+    params_len: usize,
+    retain_fields: bool,
```

Comment:
we're using this a bunch no,

I'm not a big fan of bool params for things like this because this makes it harder to understand, we could add an enum for this instead, like `FieldKind`